src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -92,22 +92,38 @@ static NamesAndTypesList getColumnsList(const ASTExpressionList * columns_defini
         }
 
         ASTPtr data_type = declare_column->data_type;
+        auto * data_type_function = data_type->as<ASTFunction>();
 
-        if (is_unsigned)
+        if (data_type_function)
         {
-            auto * data_type_function = data_type->as<ASTFunction>();
+            String type_name_upper = Poco::toUpper(data_type_function->name);
 
-            if (data_type_function)
+            if (is_unsigned)
             {
-                String type_name_upper = Poco::toUpper(data_type_function->name);
-
                 /// For example(in MySQL): CREATE TABLE test(column_name INT NOT NULL ... UNSIGNED)
-                if (type_name_upper.find("INT") != std::string::npos && !endsWith(type_name_upper, "SIGNED")
+                if (type_name_upper.find("INT") != String::npos && !endsWith(type_name_upper, "SIGNED")
                     && !endsWith(type_name_upper, "UNSIGNED"))
                     data_type_function->name = type_name_upper + " UNSIGNED";
             }
-        }
 
+            /// Transforms MySQL ENUM's list of strings to ClickHouse string-integer pairs
+            /// For example ENUM('a', 'b', 'c') -> ENUM('a'=1, 'b'=2, 'c'=3)
+            if (type_name_upper.find("ENUM") != String::npos)
+            {
+                Int64 i = 0;
+                for (ASTPtr & child : data_type_function->arguments->children)
+                {
+                    auto newChild = std::make_shared<ASTFunction>();
+                    newChild->name = "equals";
+                    auto * literal = child->as<ASTLiteral>();
+
+                    newChild->arguments = std::make_shared<ASTExpressionList>();
+                    newChild->arguments->children.push_back(std::make_shared<ASTLiteral>(literal->value.get<String>()));
```

Comment:
Lowercase this literal here.

Reply:
I thought we agreed on making values case sensitive, no?

Reply:
Sorry, my fault, yes we did :-)